### PR TITLE
fix: allow method name start with  underscore

### PR DESCRIPTION
**Pull request by hunjixin**
method name can start with under score

https://github.com/filecoin-project/FIPs/blob/0044af63ebaa669a286750d4b2ca61ffceff52c3/FRCs/frc-0042.md?plain=1#L78

### CODE REVIEW

File: frc42_dispatch/hasher/src/hash.rs [hunk 1]
```diff
@@ -50,8 +50,8 @@ pub enum MethodNameErr {
 
 #[derive(Error, PartialEq, Eq, Debug)]
 pub enum IllegalNameErr {
-    #[error("method name doesn't start with capital letter")]
-    NotCapitalStart,
+    #[error("method name doesn't start with capital letter or _")]
+    NotValidateStart,
```
Comment (alexytsu) [inline frc42_dispatch/hasher/src/hash.rs:54]:
```suggestion
    NotValidStart,
```

**Review by alexytsu — APPROVED** [review]
Thanks for this

The CI failure can be fixed by rebasing off the current `main` branch

File: frc42_dispatch/hasher/src/hash.rs [hunk 2]
```diff
@@ -110,8 +110,8 @@ fn check_method_name(method_name: &str) -> Result<(), MethodNameErr> {
 
     // Check starts with capital letter
     let first_letter = method_name.chars().next().unwrap(); // safe because we checked for empty string
-    if !first_letter.is_ascii_uppercase() {
-        return Err(IllegalNameErr::NotCapitalStart.into());
+    if !(first_letter.is_ascii_uppercase() || first_letter == '_') {
+        return Err(IllegalNameErr::NotValidateStart.into());
```
Comment (anorth) [inline frc42_dispatch/hasher/src/hash.rs:114]:
```suggestion
        return Err(IllegalNameErr:: NotValidStart.into());
```

File: frc42_dispatch/hasher/src/hash.rs [hunk 3]
```diff
@@ -175,7 +175,7 @@ mod tests {
         assert_eq!(method_hasher.method_number("").unwrap_err(), MethodNameErr::EmptyString);
         assert_eq!(
             method_hasher.method_number("invalidMethod").unwrap_err(),
-            MethodNameErr::IllegalName(IllegalNameErr::NotCapitalStart)
+            MethodNameErr::IllegalName(IllegalNameErr::NotValidateStart)
```
Comment (anorth) [inline frc42_dispatch/hasher/src/hash.rs:178]:
```suggestion
            MethodNameErr::IllegalName(IllegalNameErr::NotValidStart)
```

File: frc42_dispatch/macros/tests/naming/non-capital-start.stderr [hunk 4]
```diff
@@ -4,4 +4,4 @@ error: proc macro panicked
 5 |     let _str_hash = method_hash!("noPlaceForCamelCase");
   |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
-  = help: message: called `Result::unwrap()` on an `Err` value: IllegalName(NotCapitalStart)
+  = help: message: called `Result::unwrap()` on an `Err` value: IllegalName(NotValidateStart)
```
Comment (anorth) [inline frc42_dispatch/macros/tests/naming/non-capital-start.stderr:7]:
```suggestion
  = help: message: called `Result::unwrap()` on an `Err` value: IllegalName(NotValidStart)
```

**Review by anorth — COMMENTED** [review]
Commenting these suggestions in the home I can commit them all to this fork

**Comment by hunjixin** [comment]
sorry for reply late,  forget this, thanks for correct this pr @anorth

**Pull request merged**